django_app/redbox_app/redbox_core/models.py

```diff
@@ -42,16 +43,63 @@ class ProcessingStatusEnum(models.TextChoices):
     complete = "complete"
 
 
-# TO DO: Based on /redbox/models/file.py, but not complete
-class File(models.Model):
-    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
-    path = models.TextField(help_text="location of file")
-    name = models.TextField()
-    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, default=ProcessingStatusEnum.uploaded)
+class FileTypeEnum(models.TextChoices):
+    eml = ".eml"
+    html = ".html"
+    json = ".json"
+    md = ".md"
+    msg = ".msg"
+    rst = ".rst"
+    rtf = ".rtf"
+    txt = ".txt"
+    xml = ".xml"
+    csv = ".csv"
+    doc = ".doc"
+    docx = ".docx"
+    epub = ".epub"
+    odt = ".odt"
+    pdf = ".pdf"
+    ppt = ".ppt"
+    pptx = ".pptx"
+    tsv = ".tsv"
+    xlsx = ".xlsx"
+    htm = ".htm"
+
+
+class File(UUIDPrimaryKeyBase, TimeStampedModel):
+    processing_status = models.CharField(choices=ProcessingStatusEnum.choices, null=False, blank=False)
+    original_file = models.FileField(storage=settings.BUCKET_NAME)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    @property
+    def file_type(self):
+        name = self.original_file.name
+        return name.split(".")[-1]
+
+    @property
+    def file_url(self):
+        return self.original_file.url
+
+    @property
+    def name(self):
+        return self.original_file.name
 
     def get_processing_text(self) -> str:
```

Comment:
i wonder if this needs to be stored at all, I think it is transient?

Reply:
I wonder what transactions happen when querying the file for information, whether a request for the file is made, or if the data like file name is stored in the db?

Reply:
in the backend this is handled by https://github.com/i-dot-ai/redbox-copilot/blob/4dc8ebf96b633d2e0c2ea2a4f0387dc9c39361d9/redbox/storage/elasticsearch.py#L193

Reply:
George is your feeling for this that it should come from elsewhere in the code? A function that returns the status of the file from checking the file is in s3 and from checking the state of embed/ingest in core?

Reply:
We'd want to display the original filename on the documents page wouldn't we? Sounds like we'd need it, or are we saying we'd get this from elsewhere?

Reply:
We can have the file name, I think George was highlighting the part of the model that got the file processing status. I feel like this is a bit beyond a model function now.

I think it would be the frontend checking the status with s3, and then the frontend checking with core the status of the file in elastic

Reply:
Oh, with you now. Could we have a `processed` boolean, so we're only checking statuses for files that are still processing?